src/chompfile.rs

```diff
@@ -214,36 +213,46 @@ impl ChompTaskMaybeTemplated {
     }
     pub fn targets_vec(&self) -> Result<Vec<String>> {
         if let Some(ref target) = self.target {
-            let target_str = target.to_string();
-            validate_path(&target_str)?;
+            let target_str = resolve_path(&target.to_string());
             Ok(vec![target_str])
         } else if let Some(ref targets) = self.targets {
-            let targets = targets.clone();
-            for target in &targets {
-                validate_path(&target)?;
-            }
+            let targets = targets.clone().iter().map(|t| resolve_path(&t)).collect();
             Ok(targets)
         } else {
             Ok(vec![])
         }
     }
-    pub fn deps_vec(&self) -> Result<Vec<String>> {
+    pub fn deps_vec(&self, chompfile: &Chompfile) -> Result<Vec<String>> {
         if let Some(ref dep) = self.dep {
-            let dep_str = dep.to_string();
-            validate_path(&dep_str)?;
+            let dep_str = if chompfile.task.iter().find(|t| t.name.to_owned().unwrap_or_default().contains(dep)).is_some() || skip_special_chars(dep) {
+                dep.to_owned()
+            } else {
+                resolve_path(&dep.to_string())
+            };
             Ok(vec![dep_str])
         } else if let Some(ref deps) = self.deps {
-            let deps = deps.clone();
-            for dep in &deps {
-                validate_path(&dep)?;
-            }
+            let deps = deps
+                .clone()
+                .iter()
+                .map(|dep| {
+                    if chompfile.task.iter().find(|t| t.name.to_owned().unwrap_or_default().contains(dep)).is_some() || skip_special_chars(dep) {
```

Comment:
Maybe something more like:

```rust
// this should be a list of references, without copying
const existing_task_names = chompfile.task.filter(|&t| t.name.is_some()).map(|&&t| t.unwrap());

...

if skip_special_chars(dep) || existing_task_names.contains(dep) {
}
```

where names is then an array of string references as opposed to having copied the original task strings.